kafka-store/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -83,25 +81,22 @@
   // messages with this key
   private final K noopKey;
   private volatile long lastWrittenOffset = -1L;
-  private final SchemaRegistryConfig config;
+  private final KafkaStoreConfig config;
 
-  public KafkaStore(SchemaRegistryConfig config,
+  public KafkaStore(KafkaStoreConfig config,
                     StoreUpdateHandler<K, V> storeUpdateHandler,
                     Serializer<K, V> serializer,
                     Store<K, V> localStore,
-                    K noopKey) {
+                    K noopKey,
+                    boolean isZkSecurityEnabled) {
```

Comment:
Why make this a constructor flag rather than a config given that everything else is passed in via config?

More generally, as this becomes truly public, supported API rather than internal implementation detail, we'll need to be careful about compatibility so we'd like to make sure these interfaces, especially the constructors and most used methods, are just what we want.